### PR TITLE
[mieb] Add VLM2vec

### PR DESCRIPTION
**Pull request by isaac-chung**
## Checklist

- [ ] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 

### Adding a model checklist

 - [ ] I have filled out the ModelMeta object to the extent possible
 - [x] I have ensured that my model can be loaded using
   - [x] `mteb.get_model(model_name, revision)` and
   - [x] `mteb.get_model_meta(model_name, revision)`
 - [x] I have tested the implementation works on a representative set of tasks.

### CODE REVIEW

**Comment by isaac-chung** [comment]
@gowitheflow-1998 I wanted to run this on MSCOCOI2T and T2I but OOM on my machine. I did run a small sample to make sure the model runs but didn't manage to run the full set.

**Comment by gowitheflow-1998** [comment]
looks great, thanks for adding this! The results look great as well. Any idea what happens with the failed test?

File: mteb/models/__init__.py [hunk 1]
```diff
@@ -160,7 +161,7 @@ def model_meta_from_sentence_transformers(model: SentenceTransformer) -> ModelMe
     sentence_transformers_models,
     vista_models,
     voyage_models,
-    google_models,
```
Comment (gowitheflow-1998) [inline mteb/models/__init__.py:163]: was this accidentally removed btw?

Comment (isaac-chung) [reply mteb/models/__init__.py:163]: The item is duplicated from line 146/147.

**Review by gowitheflow-1998 — APPROVED** [review]

**Pull request merged**